samples/apps/copilot-chat-app/webapp/src/components/chat/ChatWindow.tsx

```diff
@@ -98,14 +100,28 @@ const useClasses = makeStyles({
 export const ChatWindow: React.FC = () => {
     const classes = useClasses();
     const dispatch = useAppDispatch();
+    const { instance, inProgress } = useMsal();
+    const msGraph = useGraph();
+    const chatService = new ChatService(process.env.REACT_APP_BACKEND_URI as string);
+    const { alerts } = useAppSelector((state: RootState) => state.app);
+
     const { conversations, selectedId } = useAppSelector((state: RootState) => state.conversations);
     const chatName = conversations[selectedId].title;
+    const chatUsers = conversations[selectedId].users;
+
     const [title = '', setTitle] = useState<string | undefined>(selectedId);
     const [isEditing, setIsEditing] = useState<boolean>(false);
-    const { instance, inProgress } = useMsal();
-    const { alerts } = useAppSelector((state: RootState) => state.app);
 
-    const chatService = new ChatService(process.env.REACT_APP_BACKEND_URI as string);
+    useEffect(() => {
+        if (!conversations[selectedId].userDataLoaded) {
+            void msGraph.loadUsers(chatUsers.map((user: IChatUser) => user.id)).then(() => {
+                dispatch(setUsersLoaded(selectedId));
+            });
+        }
+
+        // Limiting dependencies or else this effect is triggered on all user typing events
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectedId, chatUsers.length]);
```

Comment:
What will happen if `chatUsers.length` has changed but `conversations[selectedId].userDataLoaded` is already true? Will it load the new user data?

Reply:
This should never happen because the state for both is updated at the same time and Redux/React is single-threaded: 
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/4107e81d-1bff-45d4-9edd-a1f4e6556926)

Reply:
Ahh, Ic